schemas/common/selfservice.schema.json

```diff
@@ -0,0 +1,160 @@
+  "type": "object",
+  "definitions": {
+    "selfservice": {
+      "properties": {
+        "xdm:tenant": {
+          "name": "Tenant",
+          "description": "A description of a customer having a agreement to use a specific set of services (see `xdm:service`). A tenant will typically have a URI which can be used to reference the tenant, and should be resovable by an identity provider.",
+          "type": "string",
+          "format": "uri"
+        },
+        "xdm:applicationOwner": {
+          "name": "Application Owner",
+          "description": "An application owner, distinct from a tenant  (`xdm:tenant`), owns an application (see `xdm:application`) and manages its state. The application owner may be a individual user or roll. Typically the application owner will be a member of the tenant, however that is not a requirement.",
+          "type": "string",
+          "format": "uri"
+        },
+        "xdm:application": {
+          "name": "Application",
+          "description": "A set of code, content, and data that together implement an experience. Some applications do not allow a system integrator, partner or customer to perform customisation. Other applications may allow customisation on a per instance basis. These may be applications managed by the system integrator, partner or customer. An application be defined by a uri referencing the definition of the application.",
+          "type": "string",
+          "format": "uri"
+        },
+        "xdm:environment": {
+          "name": "Environment",
+          "description": "A tenant (see `xdm:tenant`) may work with 1..n environments where the application (see `xdm:application`) runs. Each environment serves a purpose in the CI/CD chain providing a structure whereby changes  to the application can be rolled out. Typically the owner of the application manages transitions between environments. A system integratior, partner or customer may be the owner of the application expecially where the application has  been extended and customised. An environment may have a uri referencing the definition of the environment. ",
+          "enum": [
+            "dev",
+            "qe",
+            "stage",
+            "beta",
+            "prod"
+          ],
+          "meta:enum": {
+            "dev": "A development environment for engineering teams to develop and test against prior to releasing to the QE teams.",
+            "qe": "A deployment provisioned to statisfy the Quality Engineering process",
+            "stage": "A final deployment environment before production. Typically used to perform final testing and give a reference  deployment to the last production deployment",
+            "beta": "A production deployment environment used to introduce end users of the `application` to new features.",
+            "prod": "The production deployment environment",
+            "*": "any other environment"
```

Comment:
That's not how `enum`s work. Either allow arbitrary contents or don't. Or build a property that is either `enum` or `string`